### PR TITLE
Bookstore add reducers and actions

### PR DESCRIPTION
**Pull request by jash-debug**
- [ ] Add redux Toolkit.
- [ ] Configure a redux store.
- [ ] Define a slice of state for books.
- [ ] Define a slice of state for categories.
- [ ] Update App.js with Redux.
- [ ] Update Index.js.
- [ ] Add basic styling.
- [ ] Fix linter errors.

### CODE REVIEW

File: src/redux/books/BooksSlice.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+// Action Creators
+
+const addBookAction = (book) => ({ type: ADD_BOOK, payload: { book } });
+
+const removeBookAction = (id) => ({ type: REMOVE_BOOK, payload: { id } });
+
+export {
+  ADD_BOOK, REMOVE_BOOK, addBookAction, removeBookAction,
+};
```
Comment (Zuhaib042) [inline src/redux/books/BooksSlice.js]: - [ ] Good job following the duck pattern in this file and its working but the issue is that you are required in this milestone to use the new and better way to write action creates and reducers which is through redux toolkit's feature `createSlice`. kindly consider writing the same functionality using createSlice feature of redux toolkit. you need to implement following the requirements below and you can also take help from redux toolkit official documentation.
- Define a slice of state for books that:
   -  Includes an array of books (initial state: empty array)
   -  Includes a reducer that adds a book
   -  Includes a reducer that removes a book

File: src/redux/categories/CategoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    default:
+      return state;
+  }
+}
+
+// Action Creators
+
+const checkStatusAction = () => ({ type: CHECK_STATUS });
+
+export { CHECK_STATUS, checkStatusAction };
```
Comment (Zuhaib042) [inline src/redux/categories/CategoriesSlice.js]: - [ ] Good job following the duck pattern in this file and its working but the issue is that you are required in this milestone to use the new and better way to write action creates and reducers which is through redux toolkit's feature createSlice. kindly consider writing the same functionality using createSlice feature of redux toolkit. you need to implement following the requirements below and you can also take help from redux toolkit official documentation.
- Define a slice of state for categories that:
   -  Includes an array of categories (initial state: empty array)
   -  Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @jash-debug 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ Professional Readme
✅ Descriptive PR

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @jash-debug, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Good implementation of the changes required ✔️ 
- [ ] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**